src/masternode.cpp

```diff
@@ -165,37 +160,24 @@ uint256 CMasternode::CalculateScore(const uint256& hash) const
     return (hash3 > hash2 ? hash3 - hash2 : hash2 - hash3);
 }
 
-void CMasternode::Check(bool forceCheck)
+CMasternode::state CMasternode::GetActiveState() const
 {
-    if (ShutdownRequested()) return;
-
-    LOCK(cs);
-
-    if (!forceCheck && (GetTime() - lastTimeChecked < MASTERNODE_CHECK_SECONDS)) return;
-    lastTimeChecked = GetTime();
-
-    //once spent, stop doing the checks
-    if (activeState == MASTERNODE_VIN_SPENT) return;
-
+    if (fCollateralSpent) {
```

Comment:
shouldn't be this guarded by `LOCK(cs)`?

Reply:
Or wait, seems that it's being locked in the caller side. In such case, an assertLockHeld here wouldn't hurt.

Reply:
nvm, you added it on the following commit..